x/ccv/provider/keeper/msg_server.go

```diff
@@ -309,16 +293,55 @@ func (k msgServer) SetConsumerCommissionRate(goCtx context.Context, msg *types.M
 	return &types.MsgSetConsumerCommissionRateResponse{}, nil
 }
 
-func (k msgServer) ConsumerModification(goCtx context.Context, msg *types.MsgConsumerModification) (*types.MsgConsumerModificationResponse, error) {
-	if k.GetAuthority() != msg.Authority {
-		return nil, errorsmod.Wrapf(types.ErrUnauthorized, "expected %s, got %s", k.GetAuthority(), msg.Authority)
+// RegisterConsumer registers a consumer chain
+func (k msgServer) RegisterConsumer(goCtx context.Context, msg *types.MsgRegisterConsumer) (*types.MsgRegisterConsumerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	consumerId := strconv.FormatUint(k.Keeper.FetchAndIncrementConsumerId(ctx), 10)
+
+	k.Keeper.SetConsumerIdToRegistrationRecord(ctx, consumerId, *msg.RegistrationRecord)
+	k.Keeper.SetConsumerIdToOwnerAddress(ctx, consumerId, msg.Signer)
+	k.Keeper.SetConsumerIdToPhase(ctx, consumerId, Registered)
+
+	return &types.MsgRegisterConsumerResponse{ConsumerId: consumerId}, nil
+}
+
+// InitializeConsumer initializes a consumer chain
+func (k msgServer) InitializeConsumer(goCtx context.Context, msg *types.MsgInitializeConsumer) (*types.MsgInitializeConsumerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	consumerId := msg.ConsumerId
+
+	phase, found := k.Keeper.GetConsumerIdToPhase(ctx, consumerId)
+	if !found || (phase != Registered && phase != Initialized) {
+		return nil, errorsmod.Wrapf(types.ErrInvalidPhase,
+			"chain with consumer id: %s has to be in its registered or initialized phase", consumerId)
 	}
 
+	ownerAddress, _ := k.Keeper.GetConsumerIdToOwnerAddress(ctx, consumerId)
+	if k.GetAuthority() == msg.Authority {
```

Comment:
This code is hard to read. I guess the assumption is that ownerAddress is `msg.Authority`, right?

Reply:
No, not necessarily.
```
if k.GetAuthority() == msg.Authority {
    // message is executed as part of governance proposal and hence we change the owner address
    // to be the one of the module account address
    k.Keeper.SetConsumerOwnerAddress(ctx, consumerId, k.GetAuthority())
} else if msg.Authority != ownerAddress {
    return nil, errorsmod.Wrapf(types.ErrUnauthorized, "expected owner address %s, got %s", ownerAddress, msg.Authority)
}
````
`msg.Authority` could technically be anything. If it's neither the gov address, nor the `ownerAddress` (i.e., who registered the chain), the transaction would fail. If it is the gov address, then we set the owner address to be the gov address from now on. If it's not the gov address, it **has to be** the owner of the consumer chain.

Reply:
```golang
if k.GetAuthority() == msg.Authority {
    // message is executed as part of governance proposal and hence we change the owner address
    // to be the one of the module account address
    k.Keeper.SetConsumerOwnerAddress(ctx, consumerId, k.GetAuthority())
```
Wouldn't this allow the Hub governance to take over any consumer chain?